fuzz/src/peer_crypt.rs

```diff
@@ -1,78 +1,346 @@
+				Ok(key) => { return Ok(key) },
+				_ => { }
 			}
 		}
 	}
 
-	let our_network_key = match SecretKey::from_slice(get_slice!(32)) {
-		Ok(key) => key,
-		Err(_) => return,
-	};
-	let ephemeral_key = match SecretKey::from_slice(get_slice!(32)) {
-		Ok(key) => key,
-		Err(_) => return,
-	};
+	fn generate_bool(&mut self) -> Result<bool, GeneratorFinishedError> {
+		let next_byte = self.generate_bytes(1)?;
+		Ok(next_byte[0] & 1 == 1)
+	}
+}
+
+#[derive(PartialEq)]
+struct GeneratorFinishedError { }
 
-	let mut crypter = if get_slice!(1)[0] != 0 {
-		let their_pubkey = match PublicKey::from_slice(get_slice!(33)) {
-			Ok(key) => key,
-			Err(_) => return,
+impl fmt::Debug for GeneratorFinishedError {
+	fn fmt(&self, formatter: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+		formatter.write_str("Generator out of bytes")
+	}
+}
+impl fmt::Display for GeneratorFinishedError {
+	fn fmt(&self, formatter: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+		formatter.write_str("Generator out of bytes")
+	}
+}
+impl error::Error for GeneratorFinishedError {
+	fn description(&self) -> &str { "Generator out of bytes" }
+}
+
+struct TestCtx {
+	initiator_static_public_key: PublicKey,
+	initiator_handshake: PeerHandshake,
+	responder_static_public_key: PublicKey,
+	responder_handshake: PeerHandshake,
+	act1: Vec<u8>
+}
+
+impl TestCtx {
+	fn make(generator: &mut FuzzGen) -> Result<Self, GeneratorFinishedError> {
+		let curve = secp256k1::Secp256k1::new();
+
+		// Generate needed keys for successful handshake
+		let initiator_static_private_key = generator.generate_secret_key()?;
+		let initiator_static_public_key = PublicKey::from_secret_key(&curve, &initiator_static_private_key);
+		let initiator_ephemeral_private_key = generator.generate_secret_key()?;
+		let responder_static_private_key = generator.generate_secret_key()?;
+		let responder_static_public_key = PublicKey::from_secret_key(&curve, &responder_static_private_key);
+		let responder_ephemeral_private_key = generator.generate_secret_key()?;
+
+		let mut initiator_handshake = PeerHandshake::new_outbound(&initiator_static_private_key, &responder_static_public_key, &initiator_ephemeral_private_key);
+		let act1 = initiator_handshake.set_up_outbound();
+		let responder_handshake = PeerHandshake::new_inbound(&responder_static_private_key, &responder_ephemeral_private_key);
+
+		Ok(TestCtx {
+			initiator_static_public_key,
+			initiator_handshake,
+			responder_static_public_key,
+			responder_handshake,
+			act1
+		})
+	}
+}
+
+// Common test function that sends encrypted messages between two conduits until the source data
+// runs out.
+#[inline]
+fn do_conduit_tests(generator: &mut FuzzGen, initiator_conduit: &mut Conduit, responder_conduit: &mut Conduit, failures_expected: bool) -> Result<(), GeneratorFinishedError> {
+	// Keep sending messages back and forth until the input data is consumed
+	loop {
+		// Randomly generate message length
+		let msg_len_raw = generator.generate_bytes(1)?;
```

Comment:
We are bounded to generating 256-byte length random message ?  I would push until a `u16`, maybe bound few bits low if you think that performance will be too bad.

Reply:
This decision was based on the value of the input fuzzing. With an input size of 8k, a smaller value here gives us more iterations in each simulation. You have more experience with what is more useful for tests. Just let me know!

Reply:
Right, it's just two different axis which should be covered : a) number of iterations b) size of fuzzing inputs, the latter letting potentially us discovering memory-DoS. 

Let's talk about this latter, not a reason to withhold.